src/NexusMods.App.UI/Pages/TextEdit/TextEditorPageViewModel.cs

```diff
@@ -0,0 +1,145 @@
+
+            await using var stream = await fileStore.GetFileStream(fileHash);
+            using var reader = new StreamReader(stream, Encoding.UTF8);
+            var contents = await reader.ReadToEndAsync();
+
+            return (context, contents);
+        });
+
+        var canSaveObservable = this.WhenAnyValue(
+            vm => vm.Document,
+            vm => vm.IsModified,
+            (document, isModified) => document is not null && isModified
+        );
+
+        SaveCommand = ReactiveCommand.CreateFromTask(async () =>
+        {
+            var fileId = Context!.FileId;
+            var filePath = Context!.FilePath;
+
+            var text = Document?.Text ?? string.Empty;
+
+            // hash and store the new contents
+            var bytes = Encoding.UTF8.GetBytes(text);
+            var hash = Hash.From(XxHash64Algorithm.HashBytes(bytes));
+            var size = Size.From((ulong)bytes.Length);
+
+            var ms = new MemoryStream(bytes, writable: false);
+            var streamFactory = new MemoryStreamFactory(filePath.Path, ms);
+
+            await fileStore.BackupFiles([new ArchivedFileEntry(streamFactory, hash, size)]);
+
+            // update the file
+            var db = connection.Db;
+            var storedFile = db.Get<StoredFile.Model>(fileId.Value);
+
+            using (var tx = connection.BeginTransaction())
+            {
+                tx.Add(storedFile.Id, StoredFile.Hash, hash);
+                tx.Add(storedFile.Id, StoredFile.Size, size);
+                storedFile.Remap<File.Model>().Mod.Revise(tx);
+                await tx.Commit();
+            }
+
+            IsModified = false;
+        }, canSaveObservable);
+
+        this.WhenActivated(disposables =>
+        {
+            var serialDisposable = new SerialDisposable();
+            serialDisposable.DisposeWith(disposables);
+
+            repository.Revisions(Context!.FileId.Value);
+
+            this.WhenAnyValue(vm => vm.Context)
+                .Do(context =>
+                {
+                    if (context is null)
+                    {
+                        serialDisposable.Disposable = null;
+                        return;
+                    }
+
+                    serialDisposable.Disposable = repository.Revisions(context.FileId.Value)
+                        .Select(_ => context)
+                        .OffUi()
+                        .InvokeCommand(_loadFileCommand);
+                })
+                .WhereNotNull()
+                .OffUi()
+                .InvokeCommand(_loadFileCommand)
+                .DisposeWith(disposables);
+
+            this.WhenAnyObservable(vm => vm._loadFileCommand)
+                .OnUI()
+                .SubscribeWithErrorLogging(output =>
+                {
+                    var (context, contents) = output;
+                    var filePath = context.FilePath;
+                    TabTitle = filePath.Path.ToString();
```

Comment:
Tab name should still be FileName. It is convenient to have access to the full for any future changes.

Reply:
If you hover over it you get the full path, so I guess it isn't that bad
![image](https://github.com/Nexus-Mods/NexusMods.App/assets/26797547/36ef7be4-83d0-4f4d-a8d8-53b9e7a41b59)